src/Tmds.Ssh/SftpChannel.cs

```diff
@@ -1135,7 +1163,7 @@ async ValueTask CopyBuffer(ValueTask previousCopy, long fileOffset, int length)
                     await previousCopy.ConfigureAwait(false);
                 }
 
-                localFile.Write(buffer.AsSpan(0, length));
+                dstStream.Write(buffer.AsSpan(0, length));
```

Comment:
Higher up, when `stream` got assigned, can you add:
```cs
bool writeSync = stream is FileStream or MemoryStream;
```
and then here:
```cs
if (writeSync)
{
    stream.Write(buffer.AsSpan(0, length));
}
else
{
    await stream.WriteAsync(buffer.AsMemory(0, length), cancellationToken);
}
```

Reply:
i thought about exactly that but i think the fileOffset is need to the destination or else everything will be messed up, am i wrong?

Reply:
ah yes i was thinking to remove that when async that's why i ended up needing fileOffset